### ISSUE TITLE
Persist: Split in `persist-types` and a persist runtime crate

### ISSUE THREAD

**Issue by antiguru** [MaterializeInc/materialize#9193]
Consider that the `persist` crate follows a structure similar to the `dataflow` crate. A `-types` crate defines the interfaces to the implementation, such as plans and type conversions. The runtime crate offers the functionality to interpret the instructions. Benefits are a cleaner separation and crates with clearer boundaries.

**Comment by aljoscha** [comment]
@danhhz Do we want to keep this issue open? I know you have plans (in your head at least) for what the persist crates should like in the end.

**Comment by danhhz** [comment]
Yeah, I definitely want to clean this up at some point and this issue is a nice placeholder for that

**Comment by aljoscha** [comment]
@danhhz There is now a `persist-types`, but I imagine there is more refactoring to be done?
